internal/patches.go

```diff
@@ -4,20 +4,34 @@ package internal
 import (
 	"errors"
 	"fmt"
+	"os"
 	"path"
+	"path/filepath"
 
 	yaml "gopkg.in/yaml.v3"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/kustomize/api/krusty"
 	"sigs.k8s.io/kustomize/kyaml/filesys"
+
+	"open-cluster-management.io/policy-generator-plugin/internal/types"
 )
 
+type kustomizeFile struct {
+	OpenAPI   types.Filepath `json:"openapi,omitempty" yaml:"openapi,omitempty"`
+	Patches   []Patch        `json:"patches" yaml:"patches"`
+	Resources []string       `json:"resources" yaml:"resources"`
+}
+
+type Patch struct {
+	Path string `yaml:"path,omitempty" json:"path,omitempty"`
+}
```

Comment:
Ah, right--I missed the `yaml` tags would be required for marshaling. The `Patch` struct can still be removed in favor of `types.Filepath`, though.
```suggestion
type kustomizeFile struct {
	OpenAPI   types.Filepath   `json:"openapi,omitempty" yaml:"openapi,omitempty"`
	Patches   []types.Filepath `json:"patches" yaml:"patches"`
	Resources []string         `json:"resources" yaml:"resources"`
}

```

Reply:
Fixed!